src/core/lib/iomgr/call_combiner.h

```diff
@@ -40,14 +41,23 @@
 
 extern grpc_core::TraceFlag grpc_call_combiner_trace;
 
-typedef struct {
+struct grpc_call_combiner {
+  // TSAN cannot correctly understand the synchronization primitive implemented
```

Comment:
As mentioned above, please expand this comment to document how an engineer should use this API.  Is it always needed, or only in cases where tsan can't figure something out on its own?  How can an engineer tell whether they've run into such a case or are seeing a legitimate tsan error?  If they do need to use this API, where do they put the `TsanLock()` and `TsanUnlock()` calls?  Providing an example might be really helpful here.